jwt/jwks_client.py

```diff
@@ -1,31 +1,56 @@
 import json
 import urllib.request
+from urllib.error import URLError
 from functools import lru_cache
-from typing import Any, List
+from typing import Any, List, Optional
 
 from .api_jwk import PyJWK, PyJWKSet
 from .api_jwt import decode_complete as decode_token
+from .jwk_set_cache import JWKSetCache
 from .exceptions import PyJWKClientError
 
 
 class PyJWKClient:
-    def __init__(self, uri: str, cache_keys: bool = True, max_cached_keys: int = 16):
+    def __init__(self, uri: str, cache_keys: bool = False, max_cached_keys: int = 16,
+                 cache_jwk_set: bool = True, lifespan: int = 300):
         self.uri = uri
+
+        if cache_jwk_set:
+            # Init jwt set cache with default or given lifespan.
+            # Default lifespan is 300 seconds (5 minutes).
+            self.jwk_set_cache = JWKSetCache(lifespan)
```

Comment:
do you need to validate if lifespan in >= 0? also is lifespan of 0 allowed, should there be a minimum so that the cache does not miss often?

Reply:
Good point, I added check for the lifespan value